src/NuGet.Core/NuGet.Configuration/Settings/ISettingsFile.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Configuration
+{
+    /// <summary>
+    /// Interface to expose a NuGet settings file
+    /// </summary>
+    internal interface ISettingsFile
```

Comment:
You can interact with the elements directly because the `RootElement` is exposed. I could add a `SetItemInSection` to have a similar behavior than `ISettings`, but I thought it was enough to expose the elements and let the user interact with them directly. This is an `internal` interface, therefore the only users should be inside `NuGet.Configuration`.